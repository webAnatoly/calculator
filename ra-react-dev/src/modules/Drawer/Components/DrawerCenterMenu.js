import React, { Component } from 'react'
import Divider from 'material-ui/Divider'
import List from 'material-ui/List'
import { withStyles } from 'material-ui/styles'
import { styles } from '../theme'
import DrawerMenuItem from './DrawerMenuItem'

export class CenterBlock extends Component {
  render () {
    const { classes, opened, toggleDrawer, items } = this.props

    return (
      <div className={classes.flexCenter}>
        <List
          className={classes.centerList}
          onClick={() => (opened ? toggleDrawer() : null)}
        >
          {items.map(item => (
            <DrawerMenuItem key={`center_${item.label}`} {...item} />
          ))}
        </List>
        <Divider />
      </div>
    )
  }
}

export default withStyles(styles, { withTheme: true })(CenterBlock)
