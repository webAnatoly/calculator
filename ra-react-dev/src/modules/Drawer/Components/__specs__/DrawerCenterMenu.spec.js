import React from 'react'
import { shallow } from 'enzyme'
import { CenterBlock } from '../DrawerCenterMenu'

describe('DrawerCenterMenu', () => {
  let toggleDrawer, items, opened, wrapper, classes
  beforeEach(() => {
    toggleDrawer = jest.fn()
    items = [{ label: 'testItem1' }]
    classes = {
      flexCenter: 'flexCenter',
      centerList: 'centerList'
    }
    opened = true
    wrapper = shallow(
      <CenterBlock
        opened={opened}
        classes={classes}
        toggleDrawer={toggleDrawer}
        items={items}
      />
    )
  })
  it('should renders as expected', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should handle onClick()', () => {
    wrapper
      .childAt(0)
      .props()
      .onClick()
    expect(toggleDrawer.mock.calls.length).toEqual(1)
    wrapper.setProps({ opened: false })
    wrapper
      .childAt(0)
      .props()
      .onClick()
    expect(toggleDrawer.mock.calls.length).toEqual(1)
  })

  it('should render DrawerCenterMenu with label', () => {
    expect(
      wrapper
        .childAt(0)
        .childAt(0)
        .props().label
    ).toEqual(items[0].label)
  })

  it('should render elements with defined classes', () => {
    expect(wrapper.props().className).toEqual(classes.flexCenter)
    expect(wrapper.childAt(0).props().className).toEqual(classes.centerList)
  })
})
