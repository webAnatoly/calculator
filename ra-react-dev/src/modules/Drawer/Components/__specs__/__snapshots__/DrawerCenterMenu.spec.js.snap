// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerCenterMenu should renders as expected 1`] = `
<div
  className="flexCenter"
>
  <withStyles(List)
    className="centerList"
    onClick={[Function]}
  >
    <DrawerMenuItem
      key="center_testItem1"
      label="testItem1"
    />
  </withStyles(List)>
  <withStyles(Divider) />
</div>
`;
